test/runnerComponent/test.js

```diff
@@ -170,6 +182,54 @@ define([
             .on('destroy', ready);
     });
 
+    QUnit.test('Get selected providers', assert => {
+        const ready = assert.async();
+        const container = document.getElementById('fixture-init');
+
+        assert.expect(4);
+
+        const testConfig = Object.assign(sampleConfig, {
+            provider: {
+                communicator: 'request'
+            },
+            providers: {
+                runner: {
+                    id: 'mock-runner',
+                    module: 'taoTests/test/runner/mocks/mockRunnerProvider',
+                    category: 'runner'
+                },
+                communicator: {
+                    id: 'request',
+                    module: 'core/communicator/request',
+                    category: 'request'
+                },
+                proxy: [{
+                    id: 'mock-proxy1',
+                    module: 'taoTests/test/runner/mocks/mockProxyProvider',
+                    category: 'proxy'
+                }, {
+                    id: 'mock-proxy2',
+                    module: 'taoTests/test/runner/mocks/mockProxyProvider',
+                    category: 'proxy'
+                }]
```

Comment:
Maybe you should extend the test with a *plugin* as well to test *plugin* `providerType` is not selected. Because the second part of the condition (https://github.com/oat-sa/tao-test-runner-fe/pull/9/files#diff-9971dc50b3a9703eeba03d96292b8c33R138) is never tested and if I remove it, all tests are pass.